src/compiler/llvm-to-backend/host/LLVMToHost.cpp

```diff
@@ -91,57 +88,71 @@ bool LLVMToHostTranslator::toBackendFlavor(llvm::Module &M, PassHandler &PH) {
     return false;
 
   llvm::ModulePassManager MPM;
-  PH.ModuleAnalysisManager->clear(); // for some reason we need to reset the analyses... otherwise
-                                     // we get a crash at IPSCCP
+  // TODO: Clean me up.
+  // PH.ModuleAnalysisManager->clear(); // for some reason we need to reset the analyses... otherwise
+  //                                    // we get a crash at IPSCCP
+
+  llvm::LoopAnalysisManager LAM;
+  llvm::FunctionAnalysisManager FAM;
+  llvm::CGSCCAnalysisManager CGAM;
+  llvm::ModuleAnalysisManager MAM;
 
   PH.PassBuilder->registerAnalysisRegistrationCallback([](llvm::ModuleAnalysisManager &MAM) {
     MAM.registerPass([] { return SplitterAnnotationAnalysis{}; });
   });
-  PH.PassBuilder->registerModuleAnalyses(*PH.ModuleAnalysisManager);
+  // PH.PassBuilder->registerModuleAnalyses(*PH.ModuleAnalysisManager);
+
+  PH.PassBuilder->registerModuleAnalyses(MAM);
+  PH.PassBuilder->registerCGSCCAnalyses(CGAM);
+  PH.PassBuilder->registerFunctionAnalyses(FAM);
+  PH.PassBuilder->registerLoopAnalyses(LAM);
+  PH.PassBuilder->crossRegisterProxies(LAM, FAM, CGAM, MAM);
```

Comment:
The `PassBuilder` from `PH` should already have these things registered from here:
https://github.com/AdaptiveCpp/AdaptiveCpp/blob/ba3fb96fbeb4a00929f4f93b57f0831e44ed903a/include/hipSYCL/compiler/llvm-to-backend/Utils.hpp#L89

So doing it again seems wrong. Why is it necessary? If something is missing and not set up, we should probably do it there.

Reply:
Will have to test later.. this was one of the things where reusing the MAM lead to crashes (seemingly).
Maybe one of the other, later, fixes actually solved this.